nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-standard-record-utils/src/main/java/org/apache/nifi/schema/access/HortonworksAttributeSchemaReferenceStrategy.java

```diff
@@ -84,7 +85,8 @@ public RecordSchema getSchema(Map<String, String> variables, final InputStream c
         final long schemaId = Long.parseLong(schemaIdentifier);
         final int version = Integer.parseInt(schemaVersion);
```

Comment:
I think I decided to leave the attribute strategy alone to avoid creating extra complexity..

Currently the attributes approach provides a direct lookup by id+version (which would disambiguate the branch) and users are accustomed to using this strategy when they want a specific version.

If we then allow the option to use branch instead of version, the combo of id+branch would then get the latest version on the given branch, instead of a direct look up.

Also, since we now allow specifying a version with access-by-name strategy, I wonder how often the attributes approach would even be used.

If you feel strongly about it we can definitely make it work though.